src/ui/handler/touch_zoom_rotate.js

```diff
@@ -1,293 +1,260 @@
+         * when pinching in and out.
+         */
+
+        this._minDiameter = Math.min(this._minDiameter, vector.mag());
+        const circumference = Math.PI * this._minDiameter;
+        const threshold = ROTATION_THRESHOLD / circumference * 360;
+
+        const bearingDeltaSinceStart = getBearingDelta(vector, this._startVector);
+        return Math.abs(bearingDeltaSinceStart) < threshold;
+    }
+}
 
-        const {center, bearing, scale} = this._getTouchEventData(this._lastTouchEvent);
-        const around = tr.pointLocation(center);
-        const aroundPoint = tr.locationPoint(around);
+/* PITCH */
 
-        if (gestureIntent === 'rotate') {
-            tr.bearing = this._startBearing + bearing;
-        }
+function isVertical(vector) {
+    return Math.abs(vector.y) > Math.abs(vector.x);
+}
 
-        tr.zoom = tr.scaleZoom(this._startScale * scale);
-        tr.setLocationAtPoint(this._startAround, aroundPoint);
+const ALLOWED_SINGLE_TOUCH_TIME = 100;
 
-        this._map.fire(new Event(gestureIntent, {originalEvent: this._lastTouchEvent}));
-        this._map.fire(new Event('move', {originalEvent: this._lastTouchEvent}));
+export class TouchPitchHandler extends TwoTouchHandler {
 
-        this._drainInertiaBuffer();
-        this._inertia.push([browser.now(), scale, center]);
-    }
+    _valid: boolean | void;
+    _firstMove: number;
+    _lastPoints: [Point, Point];
 
-    _onEnd(e: TouchEvent) {
-        DOM.removeEventListener(window.document, 'touchmove', this._onMove, {passive: false});
-        DOM.removeEventListener(window.document, 'touchend', this._onEnd);
+    reset() {
+        super.reset();
+        this._valid = undefined;
+        delete this._firstMove;
+        delete this._lastPoints;
+    }
 
-        const gestureIntent = this._gestureIntent;
-        const startScale = this._startScale;
+    _start(points: [Point, Point]) {
+        this._lastPoints = points;
+        if (isVertical(points[0].sub(points[1]))) {
+            // fingers are more horizontal than vertical
+            this._valid = false;
 
-        if (this._frameId) {
-            this._map._cancelRenderFrame(this._frameId);
-            this._frameId = null;
         }
-        delete this._gestureIntent;
-        delete this._startScale;
-        delete this._startBearing;
-        delete this._lastTouchEvent;
+    }
 
-        if (!gestureIntent) return;
+    _move(points: [Point, Point], center: Point, e: TouchEvent) {
+        const vectorA = points[0].sub(this._lastPoints[0]);
+        const vectorB = points[1].sub(this._lastPoints[1]);
 
-        this._map.fire(new Event(`${gestureIntent}end`, {originalEvent: e}));
+        this._valid = this.gestureBeginsVertically(vectorA, vectorB, e.timeStamp);
+        if (!this._valid) return;
 
-        this._drainInertiaBuffer();
+        this._lastPoints = points;
+        this._active = true;
+        return {
+            pitchDelta: ((vectorA.y + vectorB.y) / 2) * -0.5
```

Comment:
```suggestion
            pitchDelta: -0.25 * (vectorA.y + vectorB.y);
```

Reply:
done slightly differently in b57ca8beb6a0317c00c56280ae75e6c2a8045194